floodviz/map_utils.py

```diff
@@ -115,7 +115,62 @@ def projection_info(code, url):
     return req.text
 
 
+def filter_background(bbox, bg_filename):
+    """
+    Takes bounding box and background geojson file assumed to be the US states, and outputs a geojson-like dictionary
+    containing only those features (states) whose borders at some point intersect the bounding box, OR the state that
+    completely contains the bounding box.
 
+    :param bbox: The coordinates of the bounding box
+    :param bg_filename: the name of the background file
+    :return: the features from bg_filename whose borders intersect bbox OR the feature which completely contains bbox
+    """
+    box_lon = [bbox[0], bbox[2]]
+    box_lat = [bbox[1], bbox[3]]
+    with open(bg_filename, 'r') as bg_file:
+        bg = json.load(bg_file)
+    features = bg['features']
+    in_box = []
+    for f in features:
+        starting_len = len(in_box)
+
+        # Define points for bounding box around the state.
+        feature_max_lat = float('-inf')
+        feature_max_lon = float('-inf')
+        feature_min_lat = float('inf')
+        feature_min_lon = float('inf')
+        
+        coordinates = f['geometry']['coordinates']
+
+        for group in coordinates:
+            if len(in_box) > starting_len:
+                # This feature has already been added
+                break
+            # actual points for MultiPolygons are nested one layer deeper than those for polygons
+            if f['geometry']['type'] == 'MultiPolygon':
+                group = group[0]
```

Comment:
I'd prefer that too, but then I would end up with
```
if multipolygon:
    for pair in group[0]:
         . . .
else:
    for pair in group:
         . . .
```
And I think that's worse.

I could look at factoring out the contents of the loop into a new function, but I think I tried that earlier and gave up because those operations depend on too many variables set in the outer loops.